babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -431,6 +431,60 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testDateTimePrimaryColumn() {
```

Comment:
I think we can remove this test case because it doesn't really have to do with this change and other existing tests should break if generic identifier parsing were broken by this change.

Reply:
Make sense. What do you feel about the testDateTimePrimaryLiteral and the testDateTimePrimaryBuiltInFunction test case? These grammars are supported by the base parser too, maybe I should remove them too?

Reply:
Up to you. That's existing functionality and so should already be covered by existing tests. However, at least those are closely related to the change you're making, so I could see keeping them here as well.

Reply:
Changes made in https://github.com/googleinterns/calcite/pull/51/commits/afb4b0caf81b83fa6fec7980e56781d5fb64ec69